### PR TITLE
Support for time dimension (C125)

### PR DESCRIPTION
**Pull request by offtherailz**
## Description
Support for time dimension (porting of #2967)
support for issue [3](https://github.com/geosolutions-it/austrocontrol-C125/issues/3)

### CODE REVIEW

File: web/client/reducers/layers.js [hunk 1]
```diff
@@ -102,15 +102,24 @@ function layers(state = [], action) {
             });
             return assign({}, state, {refreshing: newLayers});
         }
+        case CHANGE_LAYER_PARAMS:
```
Comment (mbarto) [inline web/client/reducers/layers.js]: So, is CHANGE_LAYER_PARAMS just an alias of CHANGE_LAYER_PROPERTIES ?

File: web/client/epics/layers.js [hunk 2]
```diff
@@ -86,7 +89,26 @@ const refresh = action$ =>
...
         });
 
-
+/**
+ * Update dimension to all layers that have that dimension set, or for the layers indicated in the action.
+ * @memberof epics.layers
+ * @param {external:Observable} action$ manages `UPDATE_LAYERS_DIMENSION`
+ * @return {external:Observable}
+ */
+const updateDimension = (action$, {getState = () => {}} = {}) =>
```
Comment (mbarto) [inline web/client/epics/layers.js]: What is the side effect here?

File: web/client/reducers/layers.js (hunk 1 above)
Comment (offtherailz) [reply web/client/reducers/layers.js]: Yes, the action creators are different in terms of entries they add to the action object, and this is the reason of different . 
I don't know if it is easier to use the same action name with 2 different action creators or use 2 different names. Do you think I should change it?

File: web/client/epics/layers.js (hunk 2 above)
Comment (offtherailz) [reply web/client/epics/layers.js]: What do you mean? Triggers a changeLayerParams for the layers passed as property or all the layer that have the dimension, if layers array is empty.
Notice that I have extended CHANGE_LAYER_PARAMS (PROPERTIES) to accept 1 string or an array of string for layers to change, to make multiple updates easier.

File: web/client/plugins/AutoMapUpdate.jsx [hunk 3]
```diff
@@ -78,5 +77,5 @@ const AutoMapUpdatePlugin = connect(autoMapUpdateSelector, {
 module.exports = {
     AutoMapUpdatePlugin,
     reducers: {},
-    epics: {manageAutoMapUpdate, refresh}
```
Comment (MV88) [inline web/client/plugins/AutoMapUpdate.jsx]: no longer needed this?

Comment (offtherailz) [reply web/client/plugins/AutoMapUpdate.jsx]: No, refresh is now included in defaults

**Pull request merged**